zigzag/datatypes.py

```diff
@@ -9,13 +9,20 @@ class OperandABC(metaclass=ABCMeta):
     """! Abstract Base Class for all dimension- and operand-like classes"""
 
     def __init__(self, name: str):
-        self.name = name
+        self.__name = name
+        self.__hash = hash(name) ^ hash(type(self))
+
+    @property
+    def name(self):
+        """Protect the class variable from reassignment (as this would invalidate the stored hash value)"""
```

Comment:
Frozen dataclasses might be interesting for this.

https://docs.python.org/3/library/dataclasses.html#frozen-instances

Reply:
On first sight, this doesn't seem to increase hash speed. Am I missing something?